### PR TITLE
Enforce region

### PR DESCRIPTION
**Pull request by samanvp**
After adding `--regions` flag to [piplines-tools](https://github.com/googlegenomics/pipelines-tools/pull/106) we can enforce providing `--region` flag as documented in #525. 
This PR also removes `zone` from our documentations and integration tests' json files to reduce the complexity and avoid any possible conflicts between `region` and `zone`.

### CODE REVIEW

File: docs/setting_region.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Setting GCP region
```
Comment (tneymanov) [inline docs/setting_region.md]: As discussed offline, lets add info from [this page](https://cloud.google.com/dataflow/docs/concepts/regional-endpoints), to guide users into using the same region as their GCS bucket/BigQuery dataset, if they are not multi-regional.

File: README.md [hunk 2]
```diff
@@ -165,11 +165,12 @@ The BigQuery to VCF pipeline is used to export variants in BigQuery to one VCF f
...
 
-## Running jobs in a particular region/zone
+## Running jobs in a particular region
 
-You may need to constrain Cloud Dataflow job processing to a specific geographic
-region in support of your project’s security and compliance needs. See
-[Setting zone/region doc](docs/setting_zone_region.md).
+You must constrain Cloud Dataflow job processing to a specific geographic
+region. Setting `--region` flag is [required](https://beam.apache.org/blog/2019/08/22/beam-2.15.0.html)
+by Beam 2.16.0 in addition to supporting your project’s security and compliance needs. See
```
Comment (tneymanov) [inline README.md]: technically, by Beam **2.15.0** release

File: gcp_variant_transforms/vcf_to_bq.py [hunk 3]
```diff
@@ -204,10 +204,7 @@ def _get_input_dimensions(known_args, pipeline_args):
 
   estimate_sizes_job_name = pipeline_common.generate_unique_name(
       _ESTIMATE_SIZES_JOB_NAME)
-  if google_cloud_options.job_name:
-    google_cloud_options.job_name += '-' + estimate_sizes_job_name
-  else:
-    google_cloud_options.job_name = estimate_sizes_job_name
+  _update_google_cloud_job_name(google_cloud_options, estimate_sizes_job_name)
```
Comment (tneymanov) [inline gcp_variant_transforms/vcf_to_bq.py]: seems unrelated, but I'm ok with leaving this quality of life change. Up to you.

File: docs/setting_region.md (hunk 1 above)
Comment (samanvp) [reply docs/setting_region.md]: done.

File: gcp_variant_transforms/vcf_to_bq.py (hunk 3 above)
Comment (samanvp) [reply gcp_variant_transforms/vcf_to_bq.py]: removed.

**Comment by samanvp** [comment]
I updated 3 doc files:
* README.md
* docs/troubleshooting.md 
*  docs/setting_region.md

Please take a look and see if they read fluently.

File: README.md [hunk 4]
```diff
@@ -165,11 +169,11 @@ The BigQuery to VCF pipeline is used to export variants in BigQuery to one VCF f
 Please refer to [BigQuery to VCF pipeline](docs/bigquery_to_vcf.md) for more
 details.
 
-## Running jobs in a particular region/zone
+## Running jobs in a particular region
 
-You may need to constrain Cloud Dataflow job processing to a specific geographic
-region in support of your project’s security and compliance needs. See
-[Setting zone/region doc](docs/setting_zone_region.md).
+You must constrain Cloud Dataflow job processing to a specific geographic
```
Comment (kemp-google) [inline README.md]: How about

You must choose a geographic region for Cloud Dataflow to process your data by setting ...

File: docs/setting_region.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+temporary directory are located. More specifically, in order to run Variant
+Transforms most efficiently you should make sure all the following resources
+are located in the same region:
+* Your source bucket set by  `--input_pattern` flag.
+* Your pipeline's temporary location set by `--temp_location` flag.
+* Your output BigQuery dataset set by `--output_table` flag.
+* Your Dataflow pipeline set by `--region` flag.
+
+## Running jobs in a particular region
+Dataflow API [requires](https://beam.apache.org/blog/2019/08/22/beam-2.15.0.html)
```
Comment (kemp-google) [inline docs/setting_region.md]: I think 'the Dataflow API' everywhere

File: docs/setting_region.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+setting a [GCP
+region](https://cloud.google.com/compute/docs/regions-zones/#available) via
+`--region` flag to run. In addition to this requirment you might also
+choose to run Variant Transforms in a specific region following your project’s
+security and compliance requirements. For example, in order
+to restrict your processing job to Europe, update the region as follows:
+
+```bash
+# Dataflow API.
```
Comment (kemp-google) [inline docs/setting_region.md]: Odd comment?

File: docs/setting_region.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+COMMAND="/opt/gcp_variant_transforms/bin/vcf_to_bq ...
+
+docker run gcr.io/cloud-lifesciences/gcp-variant-transforms \
+  --project "${GOOGLE_CLOUD_PROJECT}" \
+  --region "${GOOGLE_CLOUD_REGION}" \
+  "${COMMAND}"
+```
+
+Note that values of `--project` and `--region` flags will be automatically
+passed as `COMMAND` args in [`piplines_runner.sh`](docker/pipelines_runner.sh).
```
Comment (kemp-google) [inline docs/setting_region.md]: This doesn't seem relevant - it's an implementation detail?

File: README.md [hunk 8]
```diff
@@ -165,11 +169,11 @@ The BigQuery to VCF pipeline is used to export variants in BigQuery to one VCF f
 Please refer to [BigQuery to VCF pipeline](docs/bigquery_to_vcf.md) for more
 details.
 
-## Running jobs in a particular region/zone
+## Running jobs in a particular region
```
Comment (kemp-google) [inline README.md]: Now that this is always required the heading doesn't make a lot of sense I don't think

File: README.md (hunk 4 above)
Comment (samanvp) [reply README.md]: done.

File: README.md (hunk 8 above)
Comment (samanvp) [reply README.md]: Agree, I removed this section and added the following sentence to `GOOGLE_CLOUD_REGION` item.

File: docs/setting_region.md (hunk 5 above)
Comment (samanvp) [reply docs/setting_region.md]: done.

File: docs/setting_region.md (hunk 6 above)
Comment (samanvp) [reply docs/setting_region.md]: removed.

File: docs/setting_region.md (hunk 7 above)
Comment (samanvp) [reply docs/setting_region.md]: I agree it's implementation details, however, I'd like to contrast this method of running VT vs running it directly using the source in which case `--region` must be provided as `COMMAND` arg.

So if you don't strongly object, I'd prefer to keep this sentence.

File: docs/troubleshooting.md [hunk 9]
```diff
@@ -14,13 +14,17 @@ group or file a GitHub issue if you believe that there is a bug in the pipeline.
...
-  `us-central1` region. You may change this by specifying `--region <region>`
-  or `--zone <zone>` when running the pipeline. You can check for quota issues
-  by navigating to the
-  [Compute Engine quotas page](https://console.cloud.google.com/iam-admin/quotas?service=compute.googleapis.com)
+  region running the pipeline. You need to [set a region](./setting_region.md) 
+  for running the pipeline by specifying `--region <region>`. You can check for
+  quota issues by navigating to the [Compute Engine quotas page](https://console.cloud.google.com/iam-admin/quotas?service=compute.googleapis.com)
   while the pipeline is running, which shows saturated quotas at the top of the
-  page.
+  page in red color.
```
Comment (kemp-google) [inline docs/troubleshooting.md]: "page (highlighted in red)"

perhaps

Comment (samanvp) [reply docs/troubleshooting.md]: done.

**Pull request merged**